From aafe51a1627c1f8b7fc4a28cb3e4099785fb9eab Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Thu, 12 Mar 2026 18:42:40 +0100
Subject: [PATCH] Add INSTALLER to Python metadata

Record that the Python distribution was installed by RPM.

I moved it from RPM spec file to CMake because RPM can have
a prereleae version which would not match a version in *.dist-info
directory name generated by CMake.

https://src.fedoraproject.org/rpms/libsolv/pull-request/14
https://src.fedoraproject.org/rpms/libsolv/pull-request/13
---
 bindings/python/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/bindings/python/CMakeLists.txt b/bindings/python/CMakeLists.txt
index deeb41f9..47d25c7e 100644
--- a/bindings/python/CMakeLists.txt
+++ b/bindings/python/CMakeLists.txt
@@ -28,6 +28,7 @@ MESSAGE (STATUS "Python installation dir: ${PYTHON_INSTALL_DIR}")
 MESSAGE (STATUS "Python include path: ${PYTHON_INCLUDE_PATH}")
 
 CONFIGURE_FILE (${CMAKE_CURRENT_SOURCE_DIR}/METADATA.in ${CMAKE_CURRENT_BINARY_DIR}/METADATA @ONLY)
+FILE (WRITE ${CMAKE_CURRENT_BINARY_DIR}/INSTALLER "rpm\n")
 
 ADD_CUSTOM_COMMAND (
     OUTPUT solv_python.c
@@ -45,4 +46,5 @@ TARGET_LINK_LIBRARIES (bindings_python ${LIBSOLV_BINDINGS_LIBRARIES} ${SYSTEM_LI
 
 INSTALL (TARGETS bindings_python LIBRARY DESTINATION ${PYTHON_INSTALL_DIR})
 INSTALL (FILES ${CMAKE_CURRENT_BINARY_DIR}/solv.py DESTINATION ${PYTHON_INSTALL_DIR})
+INSTALL (FILES ${CMAKE_CURRENT_BINARY_DIR}/INSTALLER DESTINATION ${PYTHON_INSTALL_DIR}/solv-${VERSION}.dist-info)
 INSTALL (FILES ${CMAKE_CURRENT_BINARY_DIR}/METADATA DESTINATION ${PYTHON_INSTALL_DIR}/solv-${VERSION}.dist-info)
-- 
2.53.0

